addons/binding/org.openhab.binding.homepilot/src/main/java/org/openhab/binding/homepilot/handler/HomePilotThingHandler.java

```diff
@@ -0,0 +1,113 @@
+                        // no state to update here
+                        // updateState(CHANNEL_POSITION, new PercentType(position));
+                    } else {
+                        int position = 0;
+                        if (command instanceof DecimalType) {
+                            position = ((DecimalType) command).intValue();
+                        } else if (command instanceof OnOffType) {
+                            position = ((PercentType) ((OnOffType) command).as(PercentType.class)).intValue();
+                        } else if (command instanceof UpDownType) {
+                            position = ((PercentType) ((UpDownType) command).as(PercentType.class)).intValue();
+                        } else {
+                            throw new IllegalStateException("unkown command type " + command.getClass());
+                        }
+                        position = position > 100 ? 100 : ((position < 0) ? 0 : position);
+                        handler.getGateway().handleSetPosition(getThing().getUID().getId(), position);
+                        updateState(CHANNEL_POSITION, new PercentType(position));
+                    }
+                    break;
+                case CHANNEL_STATE:
+                    handler.getGateway().handleSetOnOff(getThing().getUID().getId(), OnOffType.ON.equals(command));
+                    updateState(CHANNEL_STATE, OnOffType.ON.equals(command) ? OnOffType.ON : OnOffType.OFF);
+                    break;
+                default:
+                    throw new IllegalStateException("unknown channel id " + channelUID.getId() + " in " + channelUID);
+            }
+        }
+
+    }
+
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        super.handleUpdate(channelUID, newState);
+        logger.debug("handleUpdate " + channelUID + "; " + newState + "; " + getThing().getUID());
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        super.thingUpdated(thing);
+        logger.debug("thingUpdated " + thing);
+    }
+
+    void refresh() {
+        HomePilotBridgeHandler handler = (HomePilotBridgeHandler) getBridge().getHandler();
+        HomePilotDevice device = handler.getGateway().loadDevice(getThing().getUID().getId());
+        switch (getThing().getThingTypeUID().getId()) {
+            case ITEM_TYPE_SWITCH:
+                updateState(CHANNEL_STATE, device.getPosition() == 0 ? OnOffType.OFF : OnOffType.ON);
+                break;
+            case ITEM_TYPE_ROLLERSHUTTER:
+                int position = device.getPosition();
+                position = position > 100 ? 100 : ((position < 0) ? 0 : position);
+                // position must be between 0 and 100
+                try {
+                    updateState(CHANNEL_POSITION, new PercentType(position));
+                } catch (Exception e) {
+                    throw new RuntimeException("pos: '" + position + "'", e);
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.